contrib/CODINGSTYLE.md

```diff
@@ -0,0 +1,43 @@
+# Coding Style
+## Introduction
+This is the coding style guide for Modmesh.  
+* [Introduction](#introduction)
+* [Naming](#naming)
+* [Comments](#comments)
+* [Formatting](#formatting)
+* [Class](#class)
+* [Functions](#functions)
+
+### Naming
+1. The class names use Camel case, such as ```class R3DWidget;```. 
```

Comment:
`R3DWidget` is not a good example. QT related and non-QT can be different.
Also, let's clarify if we want `HTTPRequest` or `HttpReuqest`.

Reply:
> HTTPRequest

Sorry, I don't get it. 
How do we call the ```HTTPRequest``` or ```HttpReuqest``` in cpp?

Reply:
> `R3DWidget` is not a good example. QT related and non-QT can be different. Also, let's clarify if we want `HTTPRequest` or `HttpReuqest`.

By the way, do you mean that if the class use `HTTPRequest` or `HttpReuqest`, 
the class belongs to a QT-related classes?

Reply:
`R*` classes are Qt classes in modmesh.  I name them with the leading `R` because it's the next letter of `Q`.  I do not want to give them the leading letter `Q` which is already used by Qt itself.

`R3DWidget` is a modmesh-made Qt derived class.  The CamelCase rule is ambiguous for it because "3D" is an acronym for "three-dimensional" but "3" cannot be capitalized.  Both `R3DWidget` and `R3dWidget` are CamelCased.

@tigercosmos used `HTTPRequest` and `HttpRequest` to explain that you need to consider how to deal with acronyms.  The former keeps each of the letter in the acronym "HTTP" in upper-case.  The latter treats the whole acronym as a word, and makes it "Http".

Treating an acronym as a word will make naming more consistent.  So for acronyms, let's go with `HttpRequest`.